src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 4;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(FIGURE_COUNT);
+        switch (index) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(10) + 1);
```

Comment:
to optimize this code we can create separate private method `getRandomNumber()` and call it instead of `random.nextInt(10) + 1` everywhere